lib/service_variables.rb

```diff
@@ -77,6 +82,10 @@ def string_option(name, default: nil, enum: nil)
 
   def get(key)
     redis.hget(redis_hash_key, key)
+  rescue Redis::CannotConnectError => e
```

Comment:
This will not handle the error we experienced. Should instead be catching `Redis::BaseConnectionError`, see the exception hierarchy here: https://github.com/redis/redis-rb/blob/master/lib/redis/errors.rb#L22

Reply:
Changed to catch the generic `BaseConnectionError`